File: Unity Server Project/src/fr/technicalgames/network/packet/Disconnect_Client_Packet.java
package fr.technicalgames.network.packet;

import java.net.*;

import fr.technicalgames.client.*;
import fr.technicalgames.network.*;
import fr.technicalgames.network.common.*;

public class Disconnect_Client_Packet implements IPacket{
	
	public String pseudo; 

	public Disconnect_Client_Packet(){}
	
	public Disconnect_Client_Packet(String pseudo){
		this.pseudo = pseudo;
	}

	@Override
	public void read(DataBuffer data) {
		this.pseudo = data.getString();
	}

	@Override
	public void write(DataBuffer data) {
		data.put(this.pseudo);
	}

	@Override
	public void manage(Server server, DatagramSocket socket, Client client, IPacket packet) {
		server.closeConnection(client);
		server.sendToClients(packet);
	}
	
	
	
}

File: Unity network UDP/Assets/scripts/network/packet/Disconnect_Client_Packet.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

class Disconnect_Client_Packet : IPacket
{
    public string pseudo;

    public Disconnect_Client_Packet() { }

    public Disconnect_Client_Packet(string pseudo) { this.pseudo = pseudo; }

    void IPacket.read(DataBuffer data)
    {
        this.pseudo = data.getString();
    }

    void IPacket.write(DataBuffer data)
    {
        data.put(Register.getId(typeof(Disconnect_Client_Packet)));
        data.put(this.pseudo);
    }

    void IPacket.manage(byte[] data, Client client)
    {
        if (pseudo.Equals(MainClient.pseudo))
        {
            MainClient.connected = MainState_Connection_Request_Packet.DISCONNECTED;
            MainClient.client.disconnect();
        }
        else
        {

        }
    }


}
